### PR TITLE
[1.5.x]Add query object support for the service generation

### PR DESCRIPTION
**Pull request by lnash94**
Fix https://github.com/ballerina-platform/openapi-tools/issues/1384

Through this fix we generate an open record for some scenarios, this open record support will be received via this https://github.com/ballerina-platform/ballerina-standard-library/issues/4541

### CODE REVIEW

**Review by NipunaRanasinghe — APPROVED** [review]

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceDiagnosticMessages.java [hunk 1]
```diff
@@ -33,7 +33,9 @@ public enum ServiceDiagnosticMessages {
             "Query parameters with no array item type can not be mapped to Ballerina resource" +
                     " query parameters.", DiagnosticSeverity.ERROR),
     OAS_SERVICE_102("OAS_SERVICE_102",
-            "Query parameters with type '%s' can not be mapped to the Ballerina query parameters.",
+            "Type '%s' is not a valid query parameter type in Ballerina. " +
+                    "The supported types are string, int, float, boolean, decimal, array types of the aforementioned " +
+                    "types and map<json>.",
```
Comment (aneeshafedo) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceDiagnosticMessages.java:38]: Are we planning to update this message including records when the open record fix is given in update 6 ?

File: openapi-cli/src/test/resources/generators/service/ballerina/query/object_query.bal [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    # + default_query - 3. default query param with inline object
+    # + reference_query - 4. reference query param
+    # + required_nullable - 5. required query param with inline object and nullable true
+    # + optional_nullable - 6. optional query param with inline object
+    # + default_nullable - 7. default query param with inline object
+    # + reference_nullable - 8. reference_nullable query param won't consider nullable details when it has reference in swagger
+    # + return - returns can be any of following types
+    # http:Ok (search results matching criteria)
+    # http:BadRequest (bad input parameter)
+    resource function get inventory(Required_query required_query, Optional_query? optional_query, Manufacturer? reference_query, Required_nullable? required_nullable, Optional_nullable? optional_nullable, Manufacturer? reference_nullable, Default_query default_query = {"limit":"9","id":9}, Default_nullable default_nullable = {"limit":"9","id":9}) returns http:Ok|http:BadRequest {
```
Comment (aneeshafedo) [inline openapi-cli/src/test/resources/generators/service/ballerina/query/object_query.bal:19]: `Default_nullable default_nullable = {"limit":"9","id":9}` Will giving default record work in a scenario where there is a nested record ?

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceDiagnosticMessages.java (hunk 1 above)
Comment (lnash94) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceDiagnosticMessages.java:38]: As discussed, this implementation won't be excluding the code generation for open records. Ideally, it should be compatible with the HTTP module version including the [fix for open records](https://github.com/ballerina-platform/ballerina-standard-library/issues/4541).

Comment (aneeshafedo) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceDiagnosticMessages.java:38]: I meant that in this message it says that only these types `string, int, float, boolean, decimal` are supported. When are we going to include `record` in that ?

**Review by aneeshafedo — APPROVED** [review]

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceDiagnosticMessages.java (hunk 1 above)
Comment (lnash94) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceDiagnosticMessages.java:38]: Yeah, this error message aligned with the http module's existing message and I checked with them whether we need to modify it with `record`, and they informed that `map<json>` indicates the `record` behaviour so this message won't be modified.

**Pull request merged**